resharper/resharper-unity/src/HlslSupport/ShaderLabCppFileLocationHelper.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+using JetBrains.ReSharper.Plugins.Unity.ShaderLab.Psi.Parsing;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Cpp.Caches;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.HlslSupport
+{
+    public static class ShaderLabCppHelper
+    {
+        public static IEnumerable<(CppFileLocation Location, bool IsInclude)> GetCppFileLocations(
+            IPsiSourceFile sourceFile)
+        {
+            var lexer = new ShaderLabLexerGenerated(sourceFile.Document.Buffer);
+            lexer.Start();
+            bool prevCGInclude = false;
+            while (lexer.TokenType != null)
+            {
+                if (lexer.TokenType == ShaderLabTokenType.CG_INCLUDE)
+                    prevCGInclude = true;
+                else if (lexer.TokenType == ShaderLabTokenType.CG_CONTENT)
```

Comment:
Should also handle `HLSLPROGRAM`, `HLSLINCLUDE`, `GLSLPROGRAM` and `GLSLINCLUDE`.

Reply:
It is a bug, thank you!